test/AutoSubscribe.spec.tsx

```diff
@@ -387,6 +405,72 @@ describe('AutoSubscribe', function () {
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(5);
+
+        // This will still trigger a single state update, but will change nothing
+        SimpleStoreInstance.triggerArbitraryKey(undefined);
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(5);
+
+        // None of these keys should trigger any subscription changes
+        SimpleStoreInstance.setStoreData('a', 'a', 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        SimpleStoreInstance.setStoreData('a', 'b', 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        SimpleStoreInstance.setStoreData('a', TriggerKeys.First.toString(), 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+    });
+
+    it('autoSubscribeWithKey and key - test multi-@key compound key Subscriptions', () => {
+        let expectedState = 1;
+        const Component = makeComponent({ test_compoundMultiKeySub: true, ids: ['a', 'b'] });
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(10);
+
+        SimpleStoreInstance.setStoreDataForCompoundEnumKeyedSubscription(['a', 'b'], TriggerKeys.First, 1);
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(12);
+
+        SimpleStoreInstance.setStoreDataForCompoundEnumKeyedSubscription(['a', 'b'], TriggerKeys.Second, 1);
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(13);
+
+        SimpleStoreInstance.setStoreData('a', formCompoundKey('a', 'b', TriggerKeys.First), 1);
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(11);
+
+        // This will still trigger a single state update, but will change nothing
+        SimpleStoreInstance.triggerArbitraryKey(undefined);
+        expect(Component.state('stateChanges')).toEqual(++expectedState);
+        expect(Component.state('keyedDataSum')).toEqual(11);
+
+        // None of these keys should trigger any subscription changes
+        SimpleStoreInstance.setStoreData('a', 'a', 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        SimpleStoreInstance.setStoreData('a', 'b', 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        SimpleStoreInstance.setStoreData('a', TriggerKeys.First.toString(), 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+        SimpleStoreInstance.setStoreData('a', formCompoundKey('a', 'b'), 1);
+        expect(Component.state('stateChanges')).toEqual(expectedState);
+    });
```

Comment:
Test ideas:
* Auto-subscribe on the same compound key, and ensure only a single subscription happens. Similar to `'Auto-subscribe subscribes once per unique id'` and/or `'Auto-subscribe reuses subscription object'`.
* Triggering `Key_All` should call compound key subscriptions as well.
* Triggering a mix of keys and compound keys should work, e.g. `trigger([key, formCompoundKey(stuff)])`. This is mainly for future changes that could break this, since right now compound keys are translated down to regular keys internally so this should obviously work right now.
* `formCompoundKey` should respect argument order. Triggering on `[a, b]` will not affect subscriptions for `[b, a]`. Also, `WithKey` should be at the end.
* `formCompoundKey` should not remove duplicates, like `[a, a]` should not trigger `a` and `[a, a, b]` should not trigger `[a, b`]. I could imagine code to detect this to protect the user from "obvious mistakes", which is not correct. Meh, small idea.
* `Store.subscribe(callback, compoundKey)` should work as expected. Similar to `'Manual Subscription triggers'`.

Reply:
Added most of this -- skipped stuff that's only applicable if we make compound keys much more complicated.